staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -132,8 +139,15 @@ func diffMaps(original, modified map[string]interface{}, t reflect.Type, diffOpt
 	if t.Kind() == reflect.Ptr {
 		t = t.Elem()
 	}
+	// replaceKeysList will contains all the keys with non-nil value.
```

Comment:
// Including:
// - Adding fields to the patch present in modified, missing from original
// - Setting fields to the patch present in modified and original with different values
// - Delete fields present in original, missing from modified through
//   - IFF map field - set to nil in patch
//   - IFF list of maps && merge strategy - use deleteDirective for the elements
//   - IFF list of primitives && merge strategy - use parallel deletion list
//   - IFF list of maps or primitives with replace strategy (default) -  set patch value to the value in modified
// - Build retainkey directive for fields with retainkeys patch strategy
//  

TODO: Include links to documentation about directives for each directive listed